packages/dev/docs/pages/react-aria/hooks.mdx

```diff
@@ -0,0 +1,167 @@
+      /*- begin highlight -*/
+      ref={ref}
+      {...buttonProps}
+      /*- end highlight -*/
+      style={{
+        background: isPressed ? '#444' : '#666',
+        color: 'white',
+        padding: '6px 12px',
+        borderRadius: 4,
+        border: 'none'
+      }}>
+      {props.children}
+    </button>
+  );
+}
+
+<Button onPress={() => alert('Button pressed!')}>
+  Press me
+</Button>
+```
+
+Now you just need to add your own styling, and you have a fully accessible button component that works consistently
+across mouse, touch, keyboard, and screen readers with [high quality interactions](../blog/building-a-button-part-1.html)!
+See the [useButton](useButton.html) docs more examples.
+
+## Stateful components
+
+Many components are **stateless** — they display information to a user. Examples of stateless components include buttons,
+progress bars, and links. These components don't update as the user interacts with them.
+
+A **stateful** component has some state, and allows a user to interact with the component in order to update that state.
+Examples of stateful components include text fields, checkboxes, and selects.
+
+React Aria separates the state management logic into a separate hook that lives in [react-stately](../react-stately/index.html).
+The state hook holds the state for the component, and provides an interface to read and update that state. This allows
+this logic to be reused across different platforms, e.g. in react-native. Read more about this on the
+[architecture](../architecture.html) page.
+
+To build a stateful component, you'll need to install and import the corresponding state hook from react-stately.
+Then, call the state hook from your component, and pass the resulting state object to the React Aria hook.
+You can also use the state in your rendering code to determine what visual state to display.
+
+This example shows a number field component built with the [useNumberField](useNumberField.html) hook, along
+with the [useNumberFieldState](../react-stately/useNumberFieldState.html) hook from react-stately.
+
+```tsx example
+import {useNumberFieldState} from 'react-stately';
+import {useLocale, useNumberField} from 'react-aria';
+
+function NumberField(props) {
+  let {locale} = useLocale();
+  let state = useNumberFieldState({...props, locale});
+  let inputRef = React.useRef(null);
+  let {
+    labelProps,
+    groupProps,
+    inputProps,
+    incrementButtonProps,
+    decrementButtonProps
+  } = useNumberField(props, state, inputRef);
+
+  return (
+    <div>
+      <label {...labelProps}>{props.label}</label>
+      <div {...groupProps} style={{display: 'flex', gap: 4}}>
+        <Button {...decrementButtonProps}>-</Button>
+        <input {...inputProps} ref={inputRef} />
+        <Button {...incrementButtonProps}>+</Button>
+      </div>
+    </div>
+  );
+}
+
+<NumberField
+  label="Price"
+  defaultValue={6}
+  formatOptions={{
+    style: 'currency',
+    currency: 'USD'
+  }}
+/>
+```
```

Comment:
Hopefully these examples are at least a little more compelling than the previous ones (unstyled button and checkbox), which looked like just plain native HTML elements 😉.